ada/language/ast.py

```diff
@@ -519,13 +535,37 @@ def resolve_names():
         """
         return env.bind(
             Entity.children_env,
-            origin.bind(Self.origin_node,
-                        Entity.resolve_names_internal(True, LogicTrue()))
+            origin.bind(Self.origin_node, Entity.resolve_names_internal)
         )
 
-    resolve_names_from_closest_entry_point = Property(
-        Entity.parents.find(lambda p: p.xref_entry_point).resolve_names
-    )
+    @langkit_property(return_type=Bool)
+    def resolve_names_from_closest_entry_point():
+        """
+        Resolve names from the closest entry point up to this node.
+
+        .. note::
+            unlike ``resolve_names``, this will *not* trigger resolution of
+            every node with stop_resolution that lie in the sub-tree formed by
+            the closest entry point. It will only resolve those that are in the
+            path to resolving Self.
```

Comment:
I think I understand, nevertheless an example would be good here IMHO.

Reply:
Added more explanations!